grails-app/taglib/org/pih/warehouse/SelectTagLib.groovy

```diff
@@ -57,7 +57,6 @@ class SelectTagLib {
     def requisitionService
     def organizationService
 
-    @Cacheable("selectCategoryCache")
```

Comment:
I am not sure if this is a good idea. How looks the performance of this tag now?

Reply:
Yeah the better approach would be to flush the cache when we add new objects.

Reply:
The easiest way to do that would be to add @CacheFlush annotation to methods used for CRUD operations on these objects (either on controller actions or service methods).

```
@CacheFlush("selectCategoryCache")
def saveCategory = {
    ...
}
```

Another (debatably better) approach would be to use the GORM / hibernate event handlers in the domain classes to publish a cache flush event.

```
class Category { 
    // Not sure if you can do this "multi variable declaration" but ... 
    def afterInsert, afterUpdate = { 
        publishEvent(FlushCacheEvent(this))
    }
}
```
The event class
```
class FlushCacheEvent extends ApplicationEvent { 
    FlushCacheEvent(Category source) { 
        super(source)
    }
    ...
}
```
And then the handler
```
class FlushCacheEventService {

    def springcacheService

    void onApplicationEvent(FlushCacheEvent event) { 
        if (event.source instanceOf Category) {
            springCacheService.flush("selectCategoryCache")
        } 
        // or more generically
        springCacheService.flush("select${event.source.className}Cache")

    }
}
```

Reply:
I'm ok with either. If we're tight on time, go with the annotation approach.